lib/dogma/config.ex

```diff
@@ -25,17 +25,46 @@ defmodule Dogma.Config do
   defp get_rules do
     rules     = Application.get_env( :dogma, :rule_set, @default_set ).rules
     overrides = Application.get_env( :dogma, :override, %{} )
-    Enum.reduce( overrides, rules, &reconfigure_rule/2 )
+    rules_map = Enum.reduce( rules, %{}, &insert_rule/2 )
+    overrides
+    |> Enum.reduce( rules_map, &insert_rule/2 )
+    |> Dict.values
   end
 
   defp get_exclude do
     Application.get_env( :dogma, :exclude, [] )
   end
 
-  defp reconfigure_rule({rule, false}, rules) do
-    Dict.delete( rules, rule )
+
+  defp insert_rule(rule, acc) when is_map(rule) do
+    Dict.put(acc, rule.__struct__, rule)
   end
-  defp reconfigure_rule({rule, config}, rules) do
-    Dict.put( rules, rule, config )
+
+  defp insert_rule({rule_name, config}, acc)
+  when is_atom(rule_name) and is_list(config) do
```

Comment:
:+1: like this approach to updating the config style. Neat way of emitting deprecations and the whole method can just be removed at the next major version bump.

One thing I'd think about though is whether it'd be better to do this as a logger.warn rather than just an IO.put. As it is a warning not just a message.